src/components/PendingDomains.js

```diff
@@ -0,0 +1,114 @@
+import React from "react";
+
+
+class PendingDomains extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      newDomain: '',
+      domains: []
+    };
+    this.addDomain = this.addDomain.bind(this);
+    this.handleApproveDomain = this.handleApproveDomain.bind(this);
+    this.handleDenyDomain = this.handleDenyDomain.bind(this);
+    this.handleNewDomainChange = this.handleNewDomainChange.bind(this);
+  }
+
+  handleApproveDomain(domain) {
+    this.props.backend.setDomainIsValid(domain, true)
+  }
+
+  handleDenyDomain(domain) {
+    this.props.backend.setDomainIsValid(domain, false)
+  }
+
+  addDomain(event) {
+    event.preventDefault()
```

Comment:
you should just be able to call setDomainIsValid, no need to go through the intermediate step

Reply:
Great point